File: ruby/statler.rb
#!/usr/bin/ruby

methods = ["mode", "ino", "dev", "nlink", "uid", "gid", "size", "atime", "mtime", "ctime"]

ARGV.each do |file|
  puts file
  statinfo=File::Stat.new(file)
  p statinfo
  methods.each do |meth|
    puts meth + " : " + statinfo.send(meth).to_s
  end
end

File: python/statler.py
#!/usr/bin/env python3

methods = [
    "st_mode",
    "st_ino",
    "st_dev",
    "st_nlink",
    "st_uid",
    "st_gid",
    "st_size",
    "st_atime",
    "st_mtime",
    "st_ctime",
]

import sys, os

for file in sys.argv[1:]:
    print(file)
    statinfo = os.stat(file)
    print(statinfo)
    for s in methods:
        print(s, ":", getattr(statinfo, s))
